src/components/PeopleTable.tsx

```diff
@@ -1,644 +1,230 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+import { Loader } from './Loader';
+import { getPeople } from '../api';
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+type Props = {
+  people: Person[];
+  setPeople: (people: Person[]) => void;
+  loading: boolean;
+  setLoading: (loading: boolean) => void;
+};
+
+type SortState = {
+  field: string | null;
+  order: 'asc' | 'desc' | null;
+};
+
+export const PeopleTable: React.FC<Props> = ({
+  people,
+  setPeople,
+  loading,
+  setLoading,
+}) => {
+  const [error, setError] = useState(false);
```

Comment:
Consider providing more detailed error information when setting the error state. This can help with debugging and provide more context to the user.